xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecDRMPRIME.cpp

```diff
@@ -173,16 +173,18 @@ void CDVDVideoCodecDRMPRIME::Register()
 
 AVCodec* CDVDVideoCodecDRMPRIME::FindDecoder(CDVDStreamInfo& hints)
 {
-  AVCodec* codec = nullptr;
-  while ((codec = av_codec_next(codec)))
+  const AVCodec* codec = nullptr;
+  void *i = 0;
+
+  while ((codec = av_codec_iterate(&i)))
   {
     if (av_codec_is_decoder(codec) && codec->id == hints.codec && codec->pix_fmts)
     {
       const AVPixelFormat* fmt = codec->pix_fmts;
       while (*fmt != AV_PIX_FMT_NONE)
       {
         if (*fmt == AV_PIX_FMT_DRM_PRIME)
-          return codec;
+          return const_cast<AVCodec*>(codec);
```

Comment:
shouldn't we fix the constness instead ?

Reply:
fixed